zerver/lib/actions.py

```diff
@@ -5409,31 +5409,29 @@ def send_change_stream_message_retention_days_notification(
 
     with override_language(stream.realm.default_language):
         if old_value == Stream.MESSAGE_RETENTION_SPECIAL_VALUES_MAP["unlimited"]:
-            notification_string = _(
-                "{user} has changed the [message retention period](/help/message-retention-policy) for this stream:\n"
-                "* **Old retention period**: Forever\n"
-                "* **New retention period**: {new_value} days\n"
-                "Messages will be automatically deleted after {new_value} days."
-            )
-            notification_string = notification_string.format(user=user_mention, new_value=new_value)
+            old_retention_period = "Forever"
+            new_retention_period = f"{new_value} days"
+            summary_line = f"Messages will be automatically deleted after {new_value} days."
         elif new_value == Stream.MESSAGE_RETENTION_SPECIAL_VALUES_MAP["unlimited"]:
-            notification_string = _(
-                "{user} has changed the [message retention period](/help/message-retention-policy) for this stream:\n"
-                "* **Old retention period**: {old_value} days\n"
-                "* **New retention period**: Forever\n"
-                "Messages will be kept **Forever** and will **NOT** get automatically deleted after {old_value} days."
-            )
-            notification_string = notification_string.format(user=user_mention, old_value=old_value)
+            old_retention_period = f"{old_value} days"
+            new_retention_period = "Forever"
+            summary_line = f"Messages will be kept **Forever** and will **NOT** get automatically deleted after {old_value} days."
```

Comment:
This refactor breaks internationalization; the string `Forever` here will be the English string, since it's not wrapped with `_()`.  I think you're correct to want to do a refactor to remove duplication here, but you need the individual pieces to be wrapped with `@_()`.

https://zulip.readthedocs.io/en/latest/translating/internationalization.html provides some background.

Reply:
Thank you for pointing that out, I have refactored `_(forever)` as you suggested. I am not entirely sure on how to test the translations because changing the language setting for organization or under personal settings didn't work.

Reply:
The only way to test them end to end before translators do work involves running `manage.py makemessages` and then putting in your own strings in the files under `locale/`